app/src/main/java/io/gnosis/safe/ui/safe/settings/SafeSettingsViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package io.gnosis.safe.ui.safe.settings
+
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import kotlinx.coroutines.flow.collect
+import javax.inject.Inject
+
+class SafeSettingsViewModel @Inject constructor(
+    repositories: Repositories,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SafeSettingsState>(appDispatchers) {
+
+    private val safeRepository = repositories.safeRepository()
+
+    override fun initialState() = SafeSettingsState.SafeLoading(null)
+
+    init {
+        safeLaunch {
+            safeRepository.activeSafeFlow().collect { safe ->
+                updateState { SafeSettingsState.SafeSettings(safe!!, null) }
+            }
+        }
+    }
+
+    fun removeSafe() {
+        safeLaunch {
+            val safe = safeRepository.getActiveSafe()!!
+            safeRepository.removeSafe(safe)
+            updateState {
+                SafeSettingsState.SafeRemoved(
+                    ViewAction.NavigateTo(
+                        SafeSettingsFragmentDirections.actionSafeSettingsFragmentToNoSafeFragment()
```

Comment:
This is good default assumed behaviour. I hope it syncs with what product decides about it xD